src/Compilers/CSharp/Portable/Binder/Binder_Crefs.cs

```diff
@@ -888,7 +888,7 @@ private Symbol ConstructWithCrefTypeParameters(int arity, TypeArgumentListSyntax
 
                 var unusedDiagnostics =
 #if DEBUG
-                    new BindingDiagnosticBag(DiagnosticBag.GetInstance());
+                    BindingDiagnosticBagFactory.New(DiagnosticBag.GetInstance());
```

Comment:
downside.  because the types were callled the same thing in both layers, we need a new name at the C# and VB layer (and cannot use `new X`) to create them.  So i switched it around to be a static-factory method.  The mtehod makes a bag, passing in the C#/VB specific functionality in the callback function.